agent/tool-scripts/postprocess/mpstat-postprocess

```diff
@@ -41,39 +41,31 @@ my %graph_type;
 
 # we'll force mpstat into a LANG=C environment: two-digit year on the first line,
 # 24-hour time with *no* AM/PM designation in the timestamp.
-# but calc_timestamp() and the TSRE below are backwards-compatible and can
-# deal with AM/PM
 # Usage example: $timestamp_ms = calc_timestamp($time);
 sub calc_timestamp {
-	my $this_time = $_[0];
-
-	if ( $prev_time eq "" ) { # if this is the first time
-		$prev_time = $this_time;
-	}
-
 	# time format: HH:MM:SS
 	# 24-hour format,*no* AM/PM designation
-	my $hourminsec = $this_time;
-	$hourminsec =~ /(\d+):\d+:\d+\s?(\w\w)?/;
+	my $hourminsec = $_[0];
+	$hourminsec =~ /(\d+):(\d+):(\d+)/;
```

Comment:
Why did the original code worry about AM/PM designations? Why does the new code *not* have to worry about that?

Reply:
I don't know why it did because the line matching for timestamps did not appear to include a match for AM/PM.